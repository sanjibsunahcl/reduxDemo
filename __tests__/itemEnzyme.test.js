import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import ItemsEnzyme from '../src/screen/itemEnzyme';

import renderer from 'react-test-renderer';

// This test just uses Jest snapshot testing
it('renders correctly, test using Jest', () => {
  renderer.create(<ItemsEnzyme />);
});

// Using Jest + Enzyme
describe('<ItemsEnzyme />', () => {
  it('renders correctly, test using Jest + Enzyme', () => {
    expect(shallow(<ItemsEnzyme />)).toMatchSnapshot();
  });
});

describe('Check view item', () => {
  test('check view', () => {
    let wrapper = shallow(<ItemsEnzyme />);
    expect(
      wrapper.findWhere(node => node.prop('testID') === 'todo-item'),
    ).toBeTruthy();
  });
});
