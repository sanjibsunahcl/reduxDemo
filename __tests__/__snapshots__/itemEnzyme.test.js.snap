// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsEnzyme /> renders correctly, test using Jest + Enzyme 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="todo-item"
>
  <Text>
    Sanjib
  </Text>
  <Text
    testID="countText"
  >
    Count : 
    0
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 18,
      }
    }
  >
    Items
  </Text>
  <Button
    onPress={[Function]}
    testID="myButton"
    title="Click"
  />
</View>
`;
